x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/grid_embeddable/types.ts

```diff
@@ -40,17 +84,46 @@ export interface DataVisualizerGridInput<T = Query> {
    * If esql:true, the index pattern is used to validate time field
    */
   indexPattern?: string;
+  /**
+   * If true, the table will try to retrieve subfield information as well based on visibleFields
+   * For example: visibleFields: ['field1', 'field2'] => will show ['field1', 'field1.keyword', 'field2', 'field2.keyword']
+   */
+  shouldGetSubfields?: boolean;
+  lastReloadRequestTime?: number;
+  onTableUpdate?: (update: Partial<DataVisualizerTableState>) => void;
+}
+
+export type ESQLDataVisualizerGridEmbeddableState = Omit<
```

Comment:
I see lots of types that contain `embeddable`. As it's no longer an embeddable, shall we rename the types?

Reply:
We will have another follow up in 8.15 that makes this an embeddable to be used in dashboard. So I'd like to keep this as is for now to reduce churn.